paypal-commercetools-events/src/service/commercetools.service.ts

```diff
@@ -30,3 +31,31 @@ export const getPaymentById = async (
     .execute();
   return response.body;
 };
+
+export const findMatchingSubscription = async (
+  apiRoot: ByProjectKeyRequestBuilder,
+  key: string,
+  topic?: string,
+  projectId?: string
+): Promise<Subscription | undefined> => {
+  const {
+    body: { results: subscriptions },
+  } = await apiRoot
+    .subscriptions()
+    .get({
+      queryArgs: {
+        where: `key = "${key}"`,
+      },
+    })
+    .execute();
+  if (!topic && !projectId) {
+    return subscriptions.length > 0 ? subscriptions[0] : undefined;
+  }
+  const fittingSubscriptions = subscriptions.filter(
+    (subscription) =>
+      subscription.destination.type === 'GoogleCloudPubSub' &&
```

Comment:
subscription.destination is repeating it would be nice to use Destructuring assignment and prevent it.